streamer_proxy/main.go

```diff
@@ -0,0 +1,92 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright 2024 FeatureForm Inc.
+//
+
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+
+	"github.com/apache/arrow/go/v17/arrow/flight"
+	"github.com/featureform/helpers"
+	"github.com/featureform/logging"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type GoProxyServer struct {
+	flight.BaseFlightServer
+	streamerAddress string
+	logger          logging.Logger
+}
+
+func (gps *GoProxyServer) DoGet(ticket *flight.Ticket, stream flight.FlightService_DoGetServer) error {
+	gps.logger.Infof("Received request, forwarding to iceberg-streamer at: %v", gps.streamerAddress)
+	insecureOption := grpc.WithTransportCredentials(insecure.NewCredentials())
+	client, err := flight.NewClientWithMiddleware(gps.streamerAddress, nil, nil, insecureOption)
+	if err != nil {
+		gps.logger.Fatalf("Failed to connect to the iceberg-streamer: %v", err)
+		return err
```

Comment:
`Fatalf` calls `os.Exit`, so this would halt the program. I imagine this is probably not what we want to do, so keep it at `Errorf`.

If it is actually what we want, would recommend deleteing `return err` to make it clear nothing is returning.

Reply:
very good catch, and yes definitely want the app to keep running since it already passed main's initialization steps.